torch/hub.py

```diff
@@ -209,6 +212,58 @@ def _get_cache_or_reload(github, force_reload, verbose=True, skip_validation=Fal
 
     return repo_dir
 
+def _get_trusted_input(repo):
+    # separate function for testing
+    response = input(
+        f"The repository {repo} does not belong to the list of trusted repositories and as such cannot be downloaded. "
+        "Do you trust this repository and wish to add it to the trusted list of repositories (y/N)?")
+    return response
+
+def _check_repo(repo, trust_repo=None, calling_fn="load"):
+    hub_dir = get_dir()
+    filepath = os.path.join(hub_dir, "trusted_list")
+
+    if not os.path.exists(filepath):
+        Path(filepath).touch()
+
+        # Initialize with all existing repos
+        repos = next(os.walk(hub_dir))[1]
+        with open(filepath, "a") as file:
+            for _repo in repos:
+                file.write(_repo + "\n")
+
+    # load list
+    with open(filepath, 'r') as file:
+        trusted_repos = tuple(line.strip() for line in file)
+    is_trusted = any(repo == trusted_repo for trusted_repo in trusted_repos)
+    is_trusted = is_trusted or any(repo.startswith(trusted_repo) for trusted_repo in _TRUSTED_REPO_PREFIXES)
+
+    # to be deprecated
+    if trust_repo is None:
+        if not is_trusted:
+            warnings.warn(
+                "You are about to download an untrusted repository. In a future release, this won't be allowed. "
+                f"To add the repository to your trusted list, change the command to {calling_fn}(..., "
+                "trust_repo=False) and a command prompt will appear asking for an explicit confirmation of trust, "
+                f"or {calling_fn}(..., trust_repo=True), which will assume that the prompt is to be answered with "
+                f"'yes'.")
```

Comment:
```suggestion
                f"or {calling_fn}(..., trust_repo=True), which will assume that the prompt is to be answered with "
                f"'yes'. You can also use {calling_fn}(..., trust_repo="check") which will only prompt for confirmation if the repo is not already trusted. This will eventually be the default behaviour").
```